### PR TITLE
mariadb 10.6.4

### PR DESCRIPTION
**Pull request by chenrui333**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by chenrui333** [comment]
```
Error: An exception occurred within a child process:
  Errno::ENOENT: No such file or directory @ rb_file_s_rename - (storage/mroonga/version, storage/mroonga/version.txt)
```

**Comment by cho-m** [comment]
I think we need to partially restore some patches, since those were for version issue.

File: Formula/mariadb.rb [hunk 1]
```diff
@@ -205,19 +188,3 @@ def plist
...
-index 555ab248751..cddb6f2f2a6 100644
---- a/storage/mroonga/CMakeLists.txt
-+++ b/storage/mroonga/CMakeLists.txt
-@@ -215,8 +215,7 @@ set(MYSQL_INCLUDE_DIRS
-   "${MYSQL_REGEX_INCLUDE_DIR}"
-   "${MYSQL_RAPIDJSON_INCLUDE_DIR}"
-   "${MYSQL_LIBBINLOGEVENTS_EXPORT_DIR}"
--  "${MYSQL_LIBBINLOGEVENTS_INCLUDE_DIR}"
--  "${MYSQL_SOURCE_DIR}")
-+  "${MYSQL_LIBBINLOGEVENTS_INCLUDE_DIR}")
```
Comment (cho-m) [inline Formula/mariadb.rb:220]: Top level VERSION file still causes issues here:
https://github.com/MariaDB/server/blob/mariadb-10.6.4/storage/mroonga/CMakeLists.txt#L211-L219
```
set(MYSQL_INCLUDE_DIRS
  "${MYSQL_BUILD_DIR}/include"
  "${MYSQL_SOURCE_DIR}/sql"
  "${MYSQL_SOURCE_DIR}/include"
  "${MYSQL_REGEX_INCLUDE_DIR}"
  "${MYSQL_RAPIDJSON_INCLUDE_DIR}"
  "${MYSQL_LIBBINLOGEVENTS_EXPORT_DIR}"
  "${MYSQL_LIBBINLOGEVENTS_INCLUDE_DIR}"
  "${MYSQL_SOURCE_DIR}")
```

File: Formula/mariadb.rb [hunk 2]
```diff
@@ -33,8 +33,8 @@ class Mariadb < Formula
     # Need patch to remove MYSQL_SOURCE_DIR from include path because it contains
     # file called VERSION.
     # https://github.com/Homebrew/homebrew-core/pull/76887#issuecomment-840851149
-    # Reported upstream at https://jira.mariadb.org/browse/MDEV-7209 - this fix can be
-    # removed once that issue is closed and the fix has been merged into a stable release.
+    # Originally reported upstream at https://jira.mariadb.org/browse/MDEV-7209,
+    # but only partially fixed.
     patch :DATA
```
Comment (carlocab) [inline Formula/mariadb.rb:38]: Can we use the upstream patch to fix this instead? https://github.com/mariadb-corporation/libmarias3/commit/c71898f82598

Or does this not apply cleanly?

Comment (cho-m) [reply Formula/mariadb.rb:38]: The issue isn't the submodule (`libmarias3`). It is how packaged `mroonga` build interacts with top-level https://github.com/MariaDB/server `VERSION` file.

`mroonga` CMake build adds `MYSQL_SOURCE_DIR == #{buildpath}` as part of includes.

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**